ui/components/ui/disclosure/disclosure.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useRef, useEffect } from 'react';
```

Comment:
This component handles the togglable dropdown ux and functionality shown below:

https://user-images.githubusercontent.com/7499938/144241214-e236f048-7206-4662-98e4-713873c6685a.mp4